oteljava/all/src/main/scala/org/typelevel/otel4s/oteljava/OtelJava.scala

```diff
@@ -97,6 +103,25 @@ object OtelJava {
       )
       .evalMap(forAsync[F])
 
+  /** Creates a [[cats.effect.Resource `Resource`]] of the automatic
+    * configuration of a Java `OpenTelemetrySdk` instance.
+    *
+    * @param customize
+    *   a function for customizing the auto-configured SDK builder
+    * @return
+    *   An [[org.typelevel.otel4s.Otel4s]] resource.
+    */
+  def autoConfigured[F[_]: LiftIO: Async](
```

Comment:
unfortunately, this doesn't allow you to provide your own `LocalContext`. I'm trying to design an API to provide that functionality, though it's proving more difficult than I first thought

Reply:
possible new API: https://github.com/NthPortal/otel4s/compare/OtelJava-autoconfig/PR...NthPortal:otel4s:OtelJava-autoconfig/R4

Reply:
I had a similar problem recently. I came up with the following solution: 

https://github.com/typelevel/otel4s/blob/0e58bd0331fda09f3c6d178d61fff7717554f91e/core/common/src/main/scala/org/typelevel/otel4s/context/ContextCarrier.scala

It picks up your own local or builds it if `IOLocal` and `LiftIO` are available in the implicit scope.

And the usage:
https://github.com/typelevel/otel4s/blob/0e58bd0331fda09f3c6d178d61fff7717554f91e/sdk/all/src/main/scala/org/typelevel/otel4s/sdk/OpenTelemetrySdk.scala#L127

https://github.com/typelevel/otel4s/blob/0e58bd0331fda09f3c6d178d61fff7717554f91e/sdk/all/src/main/scala/org/typelevel/otel4s/sdk/OpenTelemetrySdk.scala#L181-L182

I can decouple `ContextCarrier` to see how it can be used in the `OtelJava`.

Reply:
I like your approach and API a lot! perhaps once it's merged we can update all the methods in `OtelJava` (other than `local` to use it

Reply:
Sounds good, let's do it that way.